client/go/outline/dynamic_key.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2024 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package outline
+
+import (
+	"io"
+	"net/http"
+
+	"github.com/Jigsaw-Code/outline-apps/client/go/outline/platerrors"
+)
+
+// FetchDynamicKeyResult represents the result of fetching a dynamic key.
+//
+// We use a struct instead of a tuple to preserve a strongly typed error that gobind recognizes.
+type FetchDynamicKeyResult struct {
+	Key   string
+	Error *platerrors.PlatformError
+}
+
+// FetchDynamicKey fetches a dynamic key from the given URL.
+//
+// The function makes an HTTP GET request to the specified URL and returns the response body as a
+// string. If the request fails or the server returns a non-2xx status code, an error is returned.
+func FetchDynamicKey(url string) *FetchDynamicKeyResult {
```

Comment:
Call it `FetchUrl`?
Is there anything specific to dynamic keys?

Reply:
Updated to `FetchResource`